src/responder/nss/nss_iface.c

```diff
@@ -222,13 +222,15 @@ sss_nss_register_backend_iface(struct sbus_connection *conn,
     SBUS_INTERFACE(iface,
         sssd_nss_MemoryCache,
         SBUS_METHODS(
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, UpdateInitgroups, sss_nss_memorycache_update_initgroups, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateAllUsers, sss_nss_memorycache_invalidate_users, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateAllGroups, sss_nss_memorycache_invalidate_groups, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateAllInitgroups, sss_nss_memorycache_invalidate_initgroups, nss_ctx),
-            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, InvalidateGroupById, sss_nss_memorycache_invalidate_group_by_id, nss_ctx)
+            SBUS_SYNC(METHOD, sssd_nss_MemoryCache, UpdateInitgroups,
```

Comment:
Maybe it can be converted to signal, I am not sure at this moment. But as you can see in the code, it does not only create the request but it also waits for it to finish: `src/providers/data_provider/dp_target_id.c:851`. So there is an intention to wait for the update to finish before finishing the data provider lookup and going back to `cache_req` in responder.

Once this PR is merged, I would like to go through the whole interface and see what we can improve and simplify. But I think it is out of scope for this PR.